src/App.tsx

```diff
@@ -11,55 +10,80 @@ import NavbarMain from "./components/NavbarSection/NavbarMain";
 import NavbarTop from "./components/NavbarSection/NavbarTop";
 import NavbarLinks from "./components/NavbarSection/NavbarLinks";
 import Admin from "./components/Admin/Admin";
-import {Store} from "./state";
-import {Provider} from "react-redux";
-import {client} from "./data/apollo/client";
-import {ApolloProvider} from "@apollo/client";
+import {ProductActionCreator} from "./state";
+import {useDispatch} from "react-redux";
+import {useQuery} from "@apollo/client";
+import {bindActionCreators} from "redux";
+import {GET_ALL_PRODUCTS} from "./data/queries";
+import {ResetProducts} from "./state/action-creators/ProductActionCreator";
 
 const App: FC = () => {
+  const dispatch = useDispatch();
+  const {AddItem, ResetProducts } = bindActionCreators(ProductActionCreator, dispatch);
+
+  const { loading, data, error } = useQuery(
+      GET_ALL_PRODUCTS
+  );
+  if (loading) console.log('Loading...');
+  if (error) console.log(error);
+  if (!data) console.log('No data!');
+
+  useEffect(() => {
+    if(data){
+      ResetProducts();
+      data.getAllProducts.map((p: any) => {
```

Comment:
We can use Array.forEach or a simple for loop (the fastest) instead of map here.